packages/web/src/components/result/ResultList.js

```diff
@@ -1,84 +1,50 @@
 import React, { Component } from 'react';
-
-import { getClassName } from '@appbaseio/reactivecore/lib/utils/helper';
 import types from '@appbaseio/reactivecore/lib/utils/types';
 import Title from '../../styles/Title';
-import ListItem, { container, Image } from '../../styles/ListItem';
-import ReactiveList from './ReactiveList';
+import ListItem from '../../styles/ListItem';
+import ResultListImage from './addons/ResultListImage';
 
 class ResultList extends Component {
-	static generateQueryOptions = props => ReactiveList.generateQueryOptions(props);
-
-	renderAsListItem = (item, triggerClickAnalytics) => {
-		const result = this.props.renderData(item);
-
-		if (result) {
-			return (
-				<ListItem
-					key={item._id}
-					href={result.url}
-					image={!!result.image}
-					small={result.image_size === 'small'}
-					className={getClassName(this.props.innerClass, 'listItem')}
-					target={this.props.target}
-					rel={this.props.target === '_blank' ? 'noopener noreferrer' : null}
-					onClick={triggerClickAnalytics}
-					{...result.containerProps}
-				>
-					{result.image ? (
-						<Image
-							src={result.image}
-							small={result.image_size === 'small'}
-							className={getClassName(this.props.innerClass, 'image')}
-						/>
-					) : null}
-					<article>
-						{result.title
-							&& (typeof result.title === 'string' ? (
-								<Title
-									dangerouslySetInnerHTML={{
-										__html: result.title,
-									}}
-									className={getClassName(this.props.innerClass, 'title')}
-								/>
-							) : (
-								<Title className={getClassName(this.props.innerClass, 'title')}>
-									{result.title}
-								</Title>
-							))}
-						{result.description
-							&& (typeof result.description === 'string' ? (
-								<div
-									dangerouslySetInnerHTML={{
-										__html: result.description,
-									}}
-								/>
-							) : (
-								<div>{result.description}</div>
-							))}
-					</article>
-				</ListItem>
-			);
-		}
-
-		return null;
-	};
+	static Image = ResultListImage;
+	static Content = ({ children, ...props }) => <article {...props}>{children}</article>;
+	static Title = ({ children, ...props }) => <Title {...props}>{children}</Title>;
+	static Description = ({ children, ...props }) => <div {...props}>{children}</div>;
 
 	render() {
-		const { renderData, ...props } = this.props;
-
-		return <ReactiveList {...props} renderData={this.renderAsListItem} listClass={container} />;
+		const {
+			children, href, small, target, ...props
+		} = this.props;
+		let hasImage = false;
+		React.Children.forEach(this.props.children, (o) => {
```

Comment:
This can some perf issues imo - we iterate over all the elements for every item in the result-list, on every render. Ideally, you should move this code in the lifecycle method and use state to provide `hasImage`?

Or, don't compute `hasImage` here at all. Compute it in the ListItem?